src/canvas/canvas.service.ts

```diff
@@ -140,12 +140,13 @@ export class CanvasService {
   }
 
   public async readState(
+    canvasId: Uuid,
     filter: CanvasStateFilter,
   ): Promise<CanvasStateEntity> {
     const queryBuilder = this.canvasStateRepository
       .createQueryBuilder()
       .addOrderBy('"dateCreated"', 'DESC');
-    queryBuilder.andWhere({ canvasId: filter.canvasId });
+    queryBuilder.andWhere({ canvasId: canvasId });
```

Comment:
sidenote: this syntax `{ canvasId: canvasId }` can be simplified to this `{canvasId}`

```js
const a = 'foo';
const b = 42;
const c = {};
const object2 = { a: a, b: b, c: c };
const object3 = { a, b, c };
```